src/transformStateWithClones.js

```diff
@@ -7,7 +7,41 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let copyState = { ...state };
+  const versionsOfState = [];
+
+  for (const object of actions) {
+    switch (object.type) {
+      case 'addProperties':
+        const { extraData } = object;
+
+        Object.assign(copyState, extraData);
+        break;
+
+      case 'removeProperties':
+        const { keysToRemove } = object;
+
+        removeProperties(copyState, keysToRemove);
+        break;
+
+      case 'clear':
+        copyState = {};
```

Comment:
if we use `const` instead of `let` - we have to delete data in our object in a different way